src/coreclr/vm/typekey.h

```diff
@@ -142,7 +142,7 @@ class TypeKey
             SUPPORTS_DAC;
         }
         CONTRACTL_END;
-        if (m_kind == ELEMENT_TYPE_CLASS)
+        if (m_kind == ELEMENT_TYPE_CLASS || m_kind == ELEMENT_TYPE_FNPTR)
```

Comment:
This does not look right. TypeKey is discriminated union. It is not valid to access `asClass` for `ELEMENT_TYPE_FNPTR`.

Reply:
Thanks. Was correct before; statements were merged together incorrectly.

Reply:
I removed this. Not sure why I added that originally (in 7's PR).